src/iop/temperature.c

```diff
@@ -1027,6 +1050,12 @@ static gboolean draw(GtkWidget *widget, cairo_t *cr, dt_iop_module_t *self)
   p->coeffs[2] /= p->coeffs[1];
   p->coeffs[1] = 1.0;
   for(int k = 0; k < 3; k++) p->coeffs[k] = fmaxf(0.0f, fminf(8.0f, p->coeffs[k]));
+
+  // If we're in a CMYG image we need to create CMYG coeffs from the RGB ones
+  dt_iop_temperature_global_data_t *gd = (dt_iop_temperature_global_data_t *)self->data;
+  const dt_image_t *img = &self->dev->image_storage;
+  if (img->flags & DT_IMAGE_4BAYER) dt_colorspaces_rgb_to_cygm(p->coeffs, 1, gd->RGB_to_CAM);
```

Comment:
Where would you do it? Note this isn't doing color conversion on images, it's just converting the spot WB RGB coeffs to CYGM coeffs.